src/models/state/wallet/wallet_state.rs

```diff
@@ -846,6 +919,26 @@ impl WalletState {
             .flat_map(|key_type| self.get_known_spending_keys(key_type))
     }
 
+    /// Return an iterator over the next n spending keys of all applicably key
+    /// types, with derivation info.
+    ///
+    /// Specifically, return an iterator over tuples (key type, derivation
+    /// index, spending key) for the next `num_future_keys` to be derived, for
+    /// key types "Generation" and "Symmetric Key".
```

Comment:
Let's make it explicit here that this function does *not* increment the key counter in the wallet DB.